### PR TITLE
Create and update article comment

### PR DESCRIPTION
**Pull request by oladimillion**
#### What does this PR do?
Creates endpoint to comment on an article
#### Description of Task to be completed?
- create comment model
- create comment controller

Have these endpoints working.
`POST /api/v1/articles/:articleSlug/comments` 
`POST /api/v1/articles/:articleSlug/comments/:commentId/reply` 
`GET /api/v1/articles/:articleSlug/comments`
`PUT /api/v1/articles/:articleSlug/comments/:commentId/edit` 
`PUT /api/v1/articles/:articleSlug/comments/:commentId/reply/:replyCommentId/edit` 

#### How should this be manually tested?
After cloning the repo, cd into it and RUN `npm install`, set env variables, and run migration
- Using Postman, test the endpoints above.
- To test authentication, add `x-access-token: JWT TOKEN`. Get the TOKEN from signup endpoint
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160451504

### CODE REVIEW

File: server/controllers/profileController.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ export default class ProfileController {
    */
   static getProfileByUsername(req, res) {
     return Profile.findOne({
-      where: { username: req.user.username },
+      where: { username: req.params.username },
```
Comment (oladimillion) [inline server/controllers/profileController.js:87]: Getting profile by username should be done using the params. So I fix this error

File: server/controllers/profileController.js [hunk 2]
```diff
@@ -103,10 +103,10 @@ export default class ProfileController {
    * @returns {object} - status, message and profile detail
    */
   static updateProfile(req, res) {
-    const body = ProfileController.updateReqBody(req, req.user);
+    const body = ProfileController.updateReqBody(req);
 
     return Profile.findOne({
-      where: { username: req.params.username },
+      where: { username: req.user.username },
```
Comment (oladimillion) [inline server/controllers/profileController.js:109]: Getting profile of an authenticated user should be done from the decode token, not from the params. So I fix this error

File: server/models/article.js [hunk 3]
```diff
@@ -26,12 +26,17 @@ export default (sequelize, DataTypes) => {
...
     }
   }, {});
+
   Article.associate = (models) => {
     Article.belongsTo(models.User, {
       foreignKey: 'userId',
       onDelete: 'CASCADE'
     });
-
+    Article.hasMany(models.ArticleComment, {
```
Comment (oladimillion) [inline server/models/article.js:36]: Add relationship between _Article model_ and _Article comment model_

File: server/tests/article.test.js [hunk 4]
```diff
@@ -15,9 +15,9 @@ const user = {
 };
 
 const article = {
-  title: faker.lorem.sentence(),
-  body: faker.lorem.paragraph(),
-  description: faker.lorem.sentences()
+  title: 'this is article title',
+  body: 'this is article body',
+  description: 'this is article description'
```
Comment (oladimillion) [inline server/tests/article.test.js]: Faker package causes the test to fail.

**Review by onwuvic — COMMENTED** [review]
LGTM

**Review by slimsolz — COMMENTED** [review]
LGTM

**Comment by s-akinrele** [comment]
Fix your merge conflicts

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**